### PR TITLE
FIX: Valid Context arguments for Statistical Eye Reports

### PR DESCRIPTION
**Pull request by mahesh-dhanekula**
Fixed code for correct context arguments in case of multiple Tx and Rx components

### CODE REVIEW

**Comment by mahesh-dhanekula** [comment]
@Samuelopez-ansys @gmalinve Can you check help on failure of IronPython tests ?

File: pyaedt/modules/report_templates.py [hunk 1]
```diff
@@ -2680,6 +2686,29 @@ def _context(self):
...
+                if len(available_quantities) == 1 and available_quantities[0].lower() == expr.lower():
+                    found = True
+                else:
+                    sid += 1
+                    pid += 1
+                    arg[2][arg[2].index("SID") + 2] = str(sid)
+                    arg[2][arg[2].index("PID") + 2] = str(pid)
+                # Limited maximum iterations to 1000 (Too many probes to analyze even in a single design)
+                if sid > 1000:
+                    self._post.logger.error(f"Failed to find right context for expression : {self.expressions}")
```
Comment (Samuelopez-ansys) [inline pyaedt/modules/report_templates.py:2708]: @mahesh-dhanekula Here is the error. In Ironpython 2.7, f"{}" did not exist. For that reason we need to use:

"Failed to find right context for expression : {}".format(self.expressions)

Comment (mahesh-dhanekula) [reply pyaedt/modules/report_templates.py:2708]: Been a fan of these **f** strings and overlooked them, will make a commit after modification

**Comment by mahesh-dhanekula** [comment]
@Samuelopez-ansys 
When I run the Iron Python unit tests, I don't see any problem as test log says all ran Ok.\
Unittest log output
```
Ironpython Unit Tests Started

Start time: 2024-06-17 22:50:34.954000

test_run_desktop_circuit (__main__.TestIronPython) ... ok
test_run_desktop_hfss (__main__.TestIronPython) ... ok
test_run_desktop_hfss3dlayout (__main__.TestIronPython) ... ok
test_run_desktop_icepak (__main__.TestIronPython) ... ok
test_run_desktop_maxwell2d (__main__.TestIronPython) ... ok
test_run_desktop_maxwell3d (__main__.TestIronPython) ... ok
test_run_desktop_mechanical (__main__.TestIronPython) ... ok
test_run_desktop_q2d (__main__.TestIronPython) ... ok
test_run_desktop_q3d (__main__.TestIronPython) ... ok
test_run_desktop_twinbuilder (__main__.TestIronPython) ... ok

----------------------------------------------------------------------
Ran 10 tests in 22.523s

OK

<unittest.runner.TextTestResult Total Test run=10>
```
AEDT batch log output
```
 
Ansys Electronics Desktop Version 2024.1.1, Build: 2024-03-20 16:10:33
Location: C:\Program Files\AnsysEM\v241\Win64\ansysedt.exe
Batch Script: D:/Mahesh/PyCharm/PyAEDT/pyaedt/_unittest_ironpython/run_unittests.py
Starting Batch Run: 10:50:22 PM  Jun 17, 2024
Running as user : ANSYS\mdhaneku
Temp directory: C:\Users\mdhaneku\AppData\Local\Temp\3
Project directory: D:\Mahesh\aedt files
[error] Unable to locate module named: Solutions.
[error] Error in command execution
Stopping Batch Run: 10:50:58 PM  Jun 17, 2024

```

**Review by Samuelopez-ansys — APPROVED** [review]
@mahesh-dhanekula Great pull request! and high score in coverage, congrats!

**Pull request merged**